lib/spack/spack/test/cmd/ci.py

```diff
@@ -1462,6 +1462,56 @@ def fake_get_mirrors_for_spec(spec=None, full_hash_match=False,
             _validate_needs_graph(new_yaml_contents, needs_graph, False)
 
 
+def test_ci_generate_prune_untouched(tmpdir, mutable_mock_env_path,
+                                     install_mockery, mock_packages,
+                                     project_dir_env, monkeypatch):
+    """Test pipeline generation with pruning works to eliminate
+       specs that were not affected by a change"""
+    project_dir_env(tmpdir.strpath)
+    mirror_url = 'https://my.fake.mirror'
+    filename = str(tmpdir.join('spack.yaml'))
+    with open(filename, 'w') as f:
+        f.write("""\
+spack:
+  specs:
+    - archive-files
+    - callpath
+  mirrors:
+    some-mirror: {0}
+  gitlab-ci:
+    mappings:
+      - match:
+          - arch=test-debian6-core2
+        runner-attributes:
+          tags:
+            - donotcare
+          image: donotcare
+""".format(mirror_url))
+    with tmpdir.as_cwd():
+        env_cmd('create', 'test', './spack.yaml')
+        outputfile = str(tmpdir.join('.gitlab-ci.yml'))
+
+        def fake_compute_affected(r1=None, r2=None):
+            return ['libdwarf']
+
+        with ev.read('test'):
+            # This kind of pruning is enabled with the following env var
+            os.environ['SPACK_PRUNE_UNTOUCHED'] = 'TRUE'
```

Comment:
Looking around I see there's also a `working_env` fixture defined [here](https://github.com/spack/spack/blob/36fcf461d934eacf9a33b59a9097ef8be9cf6d69/lib/spack/spack/test/conftest.py#L359), which seems to be used in a lot of places.

I created a draft PR (#29807) to make use of that fixture in the ci tests.  I'll rebase it on `develop` after this one is merged if that's ok with you.  I feel it would be good to get this in asap, and remove a lot of the unnecessary workload from gitlab.

Reply:
Yeah I agree with that.

Reply:
Just needs an approving review.